CDCgov/prime-reportstream
# 13794 add radxmars hl7 java class

This PR adds a Java class to support RADx MARS HL7.

Ideally, this class should also support all other ORU-R01 HL7 when catchall mappings are updated.

Test Steps:
1. Verify the Java class is compatible with the RADx MARS spec. Deviations from the spec in terms of allowing deprecated fields or the use of forwards compatible types are expected.

## Changes
- Add a Java class based on HL7 v2.7 that makes changes needed to support RADx MARS HL7.

## Checklist

### Testing

## Linked Issues
- #13794

## To be done
- #14249
